### ISSUE TITLE
Timer keeps repeating

### ISSUE THREAD

**Issue by patrickleemsantos** [MrToph/react-native-countdown-circle#3]
The time keeps on repeating and unable to finish it.

**Comment by jan-happy** [comment]
Same here. The demo code never reaches 0, it just stays at 10.

**Comment by patrickleemsantos** [comment]
Hi jan-happy, I think what's happening here is every time the page renders, the timer keeps repeating.

**Comment by jan-happy** [comment]
That was also my guess.

**Comment by MrToph** [comment]
Yes, when one of the props changes the timer resets and starts over again. That's actually desired, if you change the seconds to some other value, it will reset it to that value.

Unfortunately, this also happens when you use the timer as in the example code, because `onTimeElapsed` is a new function for _each_ render. (It checks for strict equality.) I'll provide a fix hopefully soon.

Can you confirm that the issue does not happen when you make `onTimeElapsed` a member function of the class (and `textStyle`)?

**Comment by ckleo** [comment]
Having the same issue. Confirmed that the issue disappears when I remove both 'textStyle' and 'onTimeElapsed' from CountdownCircle.

`<CountdownCircle
                      seconds={10}
                      radius={30}
                      borderWidth={8}
                      color="#ff003f"
                      bgColor="#fff"
                  />`

**Comment by ckleo** [comment]
Ok, I worked around the issue by removing the componentWillReceiveProps function entirely. It might not work in your case though if you need to update the seconds and restart the timer.

**Comment by patrickleemsantos** [comment]
My solution was to prevent re-render when there's no update on the props.

```
shouldComponentUpdate(nextProps) {
   return !deepEqual(this.props, nextProps);
}
```

**Comment by MrToph** [comment]
Thanks for confirming the issue. I'll provide an update tomorrow

**Comment by MrToph** [comment]
Should be fixed in version `3.0.0` now
